src/common/components/dateTimePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,303 @@
+  }
+
+  return format(value, "H.m");
+}
+
+function getDateFromDateString(dateString?: string): Date | undefined {
+  if (!dateString) {
+    return;
+  }
+
+  const [D, M, Y] = dateString.split(".");
+
+  if (!D || !M || !Y) {
+    return;
+  }
+
+  return new Date(Number(Y), Number(M) - 1, Number(D));
+}
+
+function getDateFromDateAndTimeString(
+  dateString?: string,
+  timeString?: string
+): Date | undefined {
+  if (!dateString || !timeString) {
+    return;
+  }
+
+  const [D, M, Y] = dateString.split(".");
+  const [H, m] = timeString.split(":");
+  const hasAllDateParts = [D, M, Y, H, m].reduce(
+    (acc, part) => acc && Boolean(part),
+    true
+  );
+
+  if (!hasAllDateParts) {
+    return;
+  }
+
+  return new Date(Number(Y), Number(M) - 1, Number(D), Number(H), Number(m));
+}
+
+function validateDate(
+  date: Date | undefined | null,
+  dateToCompare: Date | undefined | null,
+  fnc: (date: Date, dateToCompare: Date) => boolean,
+  message: string
+): string {
+  if (!date || !dateToCompare) {
+    return "";
+  }
+
+  const isError = fnc(date, dateToCompare);
+
+  if (isError) {
+    return message;
+  }
+
+  return "";
+}
+
+function useOnOutsideClick<T extends HTMLElement>(
+  callback: () => void
+): Ref<T | null> {
+  const ref = useRef<null | T>(null);
+
+  useEffect(() => {
+    const listener = (event) => {
+      if (ref.current && !ref.current.contains(event.target)) {
+        callback();
+      }
+    };
+
+    document.addEventListener("mousedown", listener);
+    document.addEventListener("touchstart", listener);
+
+    return () => {
+      document.removeEventListener("mousedown", listener);
+      document.removeEventListener("touchstart", listener);
+    };
+  }, [ref, callback]);
+
+  return ref;
+}
+
+type Props = {
+  id: string;
+  name: string;
+  onChange?: (value: Date) => void;
+  value?: Date;
+  label: string;
+  locale?: Locale;
+  minDate?: Date;
+  maxDate?: Date;
+  minDateErrorMessage?: string;
+  maxDateErrorMessage?: string;
+};
+
+export default function DateTimePicker({
+  id,
+  name,
+  label,
+  onChange,
+  value = null,
+  locale = "fi",
+  minDate,
+  maxDate,
+  minDateErrorMessage,
+  maxDateErrorMessage,
+}: Props) {
+  const dateString = getDate(value);
+  const timeString = getTime(value);
+
+  const { t } = useTranslation("date_time_picker");
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [intermediaryDate, setIntermediaryDate] = useIntermediaryState<
+    string | null
+  >(dateString);
+  const [intermediaryTime, setIntermediaryTime] = useIntermediaryState<
+    string | null
+  >(timeString);
+  const [date, setDate] = useState<Date | null>(value);
+  const buttonRef = useRef(null);
+  const containerRef = useOnOutsideClick<HTMLDivElement>(
+    useCallback(() => setIsOpen(false), [])
+  );
+
+  const handleToggleMenu = () => {
+    setIsOpen((previousValue) => !previousValue);
+  };
+
+  const handleCloseMenuButtonClick = () => {
+    setIsOpen(false);
+  };
```

Comment:
I feel like inlining these 2 functions would make this more readable 🤔 . I personally can't see the benefit creating these handler functions. To me feels like it is just 3 more lines of code that bloats the component with no benefit.

This is also minor complaint

Reply:
I understand if it is a stylistic decision to create new functions for all handlers even when they are one liners but I think it would be better to keep thing more concise to improve readability and every abstraction add little bit more cognitive load when reasing these.

Maybe my biggest problem is that when you are trying to understand the code / component as a whole and it requires to jump around in the code even for smallest things.

Reply:
Yeah, I understand. I guess in general we differ a bit on our views when it comes to the amount of lines. I used to be a proponent of brevity, but I don't usually put that much focus on it anymore. I find that long term it's better not to risk being unclear by being concise.

I definitely can see the cognitive load thing as well, but for me it's less load to read the name of the handler instead of inferring its purpose form the actual logic. But I guess we are just different in that and I can see how it's annoying to jump around instead of reading the logic as you scroll down.

Reply:
I only prefer to be concise when it is doesn't sacrifice readability :) But this is nitpicking to be honest, sorry about that :D Still nice to have these discussions so we understand our code choices better.